crowbar_framework/app/helpers/installer/upgrades_helper.rb

```diff
@@ -0,0 +1,79 @@
+
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Installer
+  module UpgradesHelper
+    def continue_button(disabled = false)
+      if disabled
+        button_tag icon_tag(:chevron_right, t(".continue")), class: "btn btn-primary disabled"
+      else
+        button_tag icon_tag(:chevron_right, t(".continue")), class: "btn btn-primary"
+      end
+    end
+
+    def check_repos_button
+      if check_repos
+        continue_button
+      else
+        button_tag icon_tag(:refresh, t(".recheck")), class: "btn btn-primary"
+      end
+    end
+
+    def panel_body_color(boolen)
+      if boolen
+        "success"
+      else
+        "danger"
+      end
+    end
+
+    def check_ha_repo
+      return nil unless Proposal.where(barclamp: "pacemaker")
+      return false unless Crowbar::Repository.provided?("ha")
+
+      unless Crowbar::Repository.provided_and_enabled?("ha")
+        Openstack::Upgrade.enable_repos_for_feature("ha", logger)
+      end
+
+      true
+    end
+
+    def check_ceph_repo
+      return nil unless Proposal.where(barclamp: "ceph")
+      return false unless Crowbar::Repository.provided?("ceph")
+
+      unless Crowbar::Repository.provided_and_enabled?("ceph")
+        Openstack::Upgrade.enable_repos_for_feature("ceph", logger)
+      end
+
+      true
+    end
+
+    def check_repos
+      check_ha_repo != false && check_ceph_repo != false
```

Comment:
Please do it ruby conventional to suffix boolean return value methods mit `?`, beside that just check for `check_ha_repos? && check_ceph_repo?`